website/static/js/comment.js

```diff
@@ -14,6 +14,187 @@ var osfHelpers = require('js/osfHelpers');
+        });
+    },
+    matcher: function(flag, subtext, should_startWithSpace, acceptSpaceBar) {
+        acceptSpaceBar = true;
+        var _a, _y, match, regexp, space;
+        flag = flag.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&');
+        if (should_startWithSpace) {
+            flag = '(?:^|\\s)' + flag;
+        }
+        _a = decodeURI('%C3%80');
+        _y = decodeURI('%C3%BF');
+        space = acceptSpaceBar ? '\ ' : '';
+        regexp = new RegExp(flag + '([A-Za-z' + _a + '-' + _y + '0-9_' + space + '\'\.\+\-]*)$|' + flag + '([^\\x00-\\xff]*)$', 'gi');
+        match = regexp.exec(subtext.replace(/\s/g, ' '));
+        if (match) {
+            return match[2] || match[1];
+        } else {
+            return null;
+        }
+    }
+};
+
+var headerTemplate = '<div class="atwho-header">Contributors<small>&nbsp;↑&nbsp;↓&nbsp;</small></div>';
+var displayTemplate = '<li>${fullName}</li>';
+
+var at_config = {
+    at: '@',
+    headerTpl: headerTemplate,
+    insertTpl: '@${fullName}',
+    displayTpl: displayTemplate,
+    searchKey: 'fullName',
+    limit: 6,
+    callbacks: callbacks
+};
+
+var plus_config = {
+    at: '+',
+    headerTpl: headerTemplate,
+    insertTpl: '+${fullName}',
+    displayTpl: displayTemplate,
+    searchKey: 'fullName',
+    limit: 6,
+    callbacks: callbacks
+};
+
+var input = $('.atwho-input');
+var nodeId = window.nodeId;
+
+var getContributorList = function(input, nodeId) {
+    var url = osfHelpers.apiV2Url('nodes/' + nodeId + '/contributors/', {});
+    var request = osfHelpers.ajaxJSON(
+        'GET',
+        url,
+        {'isCors': true});
+    request.done(function(response) {
+        var activeContributors = response.data.filter(function(item) {
+            return item.embeds.users.data.attributes.active === true;
+        });
+        var data = activeContributors.map(function(item) {
+            return {
+                'id': item.id,
+                'name': item.embeds.users.data.attributes.given_name,
+                'fullName': item.embeds.users.data.attributes.full_name,
+                'link': item.embeds.users.data.links.html
+            };
+        });
+        // for any input areas that currently exist on page
+        input.atwho('load','@', data).atwho('load', '+', data).atwho('run');
+        // for future input areas so that data doesn't need to be reloaded
+        at_config.data = data;
+        plus_config.data = data;
+    });
+    request.fail(function(xhr, status, error) {
+        Raven.captureMessage('Error getting contributors', {
+            url: url,
+            status: status,
+            error: error
+        });
+    });
+    return request;
+};
+
+getContributorList(input, nodeId);
+input.atwho(at_config).atwho(plus_config).bind('paste', onPaste).on('focusin keyup', lastElementBr).on('focusout', onlyElementBr).keydown(function(e) {
+    if(e.which === 13 && !e.isDefaultPrevented()) {
+        onReturn(e);
+    }
+});
```

Comment:
This file is now pushing 1k LOC. Time to split it up! Let's move the At.js configuration into a separate module.